Added movement tokens for Alpha Strike in play mode

james-elliott opened the pull request:
This adds a small hex token next to pilot skill with a <?>. Clicking on it will show the appropriate movement tokens for the unit. Selecting the movement type will now show the letter and coloring of the movement token in the small hex.

TTM is updated to reflect movement. It should respect JMPS# and JMPW#.

Movement affects to firing should be respected as well. Standstill reduces by 1, jumping increases by 2, jumping jack SPA is considered as well as BA having no jump penalty.

This is how the play mode loads. Each unit is given a movement token with <?> to indicate that they haven't had an option selected yet.
![image](https://github.com/user-attachments/assets/bee85374-ade1-405a-a3f0-56424ffe6153)

Some tokens selected and one in the process of being selected. Jump is only available to units that have it as a valid option.
![image](https://github.com/user-attachments/assets/62da3045-f3e5-444c-adc4-af82a27d6cdf)

Aerospace movement token options to help with tracking flight level.
![image](https://github.com/user-attachments/assets/8c886905-0bb0-4628-90b2-de8f77cc5701)

james-elliott committed:
Added movement tokens for Alpha Strike in play mode

This adds a small hex token next to pilot skill with a <?>. Clicking on it will show the appropriate movement tokens for the unit. Selecting the movement type will now show the letter and coloring of the movement token in the small hex.

TTM is updated to reflect movement. It should respect JMPS# and JMPW#.

Movement affects to firing should be respected as well. Standstill reduces by 1, jumping increases by 2, jumping jack SPA is considered as well as BA having no jump penalty.

OftKilted commented:
Does it take into consideration that infantry / battle armor also don’t get the standstill benefit?

james-elliott committed:
Fix to-hit for BA and Infantry

BA and infantry are unaffected by attacker movement modifiers. Updated the calculations to skip modifying the to-hit for these unit types.

james-elliott commented:
> Does it take into consideration that infantry / battle armor also don’t get the standstill benefit?

It did not. Researching this also answered another question I had. Jump BA/Infantry with Jumping Jack SPA would theoretically get -1 to-hit, but pg 42 of the AS:CE is clear that they get no attacker movement modifiers at all. I pushed a commit that skips modifying the to-hit when the unit type is BA or Inf.

james-elliott committed:
Removing some cruft

Additional cleanup

hesto2 commented:
Oops didn't mean to approve this lol, I was taking a look. Apologies 😅

james-elliott committed:
Updates and clean up

Removed some extraneous changes I let slip into this PR.

Added a "grounded" option for Aerospace to indicate that they are moving along the ground on the map as per https://discord.com/channels/1307019240867364937/1335474984063078551/1335474984063078551

HeySporky commented:
I'm 50/50 on this one, as movement mode is something that I think your opponent should be able to see. Will opened discussion on Discord. Might want this as an option that can be turned on in settings.
